### DISCUSSION TITLE
Varying the reference position in `quarto-journals` template

### DISCUSSION THREAD

**Discussion by sammo3182** [quarto-dev/quarto-cli#2009] [Q&A] [answered]
Recently I asked a question about varying the reference allocation with Quarto in #1728. Then I find the problem that made the old rmarkdown tricks not work was because I used the [`article-format-template`](https://github.com/quarto-journals/article-format-template). The reference was in the right place I want when rendering the plain pdf but then moved to the very end of the document when rendering the `aft`. 

A replicable example:

```qmd
---
format:
  pdf:
    number-sections: true
  aft-pdf:
    keep-tex: true  
  aft-html: default

bibliography: bio.bib
title:  A test
---

content @Hu2020

# Reference

::: {#refs}
:::

# Online Supplementary Materials

```
[bio.zip](https://github.com/quarto-dev/quarto-cli/files/9386033/bio.zip)

See the difference of the reference when choosing the following options respectively:

![image](https://user-images.githubusercontent.com/6463211/185724871-cd89a2f3-24d7-4787-8515-d2aba80475e6.png)

How could I allow the `aft` to recognize my reference setting? Any suggestion will be very helpful.
Moreover, a more detailed instruction and examples about how to customize the template than the current [brief version](https://github.com/quarto-journals/article-format-template/blob/main/README.md) under aft repo and Quarto.org will be super welcomed---details like how to arrange the author area, showing the corresponding author, adding the key words and acknowledge, etc.

**Comment by mcanouil** [comment, marked as answer]
I used `quarto use template quarto-journals/article-format-template` and rendered the template using the default with the references at the end which gave the proper and expected results.

Then I moved the references section to the top.
It gave me the proper and expected results, which is, the references section at the top.
<img width="1624" alt="image" src="https://user-images.githubusercontent.com/8896044/185752738-d773a262-9c4a-4fc7-8ae7-1680337a1ebf.png">


> Moreover, a more detailed instruction and examples about how to customize the template than the current [brief version](https://github.com/quarto-journals/article-format-template/blob/main/README.md) under aft repo and Quarto.org will be super welcomed---details like how to arrange the author area, showing the corresponding author, adding the key words and acknowledge, etc.

This will require you to have some knowledge about LaTeX first, and how document classes work.
It's basically not Quarto per se. The format extension allows you to define or use LaTeX document classes and partials (pieces of LaTeX with a single goal).
I suggest for you to also take a look at the actual PDF formats in the name organisation https://github.com/quarto-journals and see how the different LaTeX parts come together.

For example in the template:

- A partial to handle authors metadata <https://github.com/quarto-journals/article-format-template/blob/main/_extensions/aft/partials/_authors.tex>
- A partial to handle the title block which includes authors <https://github.com/quarto-journals/article-format-template/blob/main/_extensions/aft/partials/title.tex>: `\author{${ by-author:_authors.tex()[\and ] }}`

**Reply by sammo3182** [reply]
@mcanouil Thank you for the answer and information! Per the info about Latex, thank you so very much! I'll look into it!

Yet, from the screenshot, it seems that you are still compiling a regular quarto pdf instead of an aft. You can see that from the outlet of the author area. The aft presents the affiliation and orcid etc. You specified them in the YAML but not shown in the output preview. Could you try to just click the `render aft-pdf` and see what comes out?

**Reply by mcanouil** [reply]
My bad, I took the wrong screenshot (too much tests):
<img width="1624" alt="image" src="https://user-images.githubusercontent.com/8896044/185774309-a2192490-3c60-45cd-9fa7-41b8d19d16be.png">
Indeed, there is something fishy for the references.

**Reply by mcanouil** [reply]
@jjallaire It appears there is an issue (ot at least an undocumented behaviour in <https://quarto.org/docs/authoring/footnotes-and-citations.html#bibliography-generation>) when using `cite-method: natbib` (probably the same thing for `cite-method: biblatex`), which can be replicated also with other formats such as https://github.com/quarto-journals/jss

- default cite-method (_i.e._, Pandoc citeproc)
  ```tex
  \hypertarget{references}{%
  \subsubsection*{References}\label{references}}
  \addcontentsline{toc}{subsubsection}{References}
  
  \hypertarget{refs}{}
  \begin{CSLReferences}{1}{0}
  \leavevmode\vadjust pre{\hypertarget{ref-CameronTrivedi2013}{}}%
  Cameron, A. Colin, and Pravin K. Trivedi. 2013. \emph{Regression
  Analysis of Count Data}. 2nd ed. Cambridge: Cambridge University Press.
  
  \end{CSLReferences}
  ```
- natbib
  ```tex
  \hypertarget{references}{%
  \subsubsection*{References}\label{references}}
  \addcontentsline{toc}{subsubsection}{References}
  
  \hypertarget{refs}{}
  \begin{CSLReferences}{0}{0}
  \end{CSLReferences}
  ```
  and at the end
  ```tex
  \bibliography{bibliography.bib}
  \end{document}
  ```

The LaTeX partial `biblio.tex` is added to the end, no matter if `refs` div is set and where it is set.
It's possible to manually move it (and remove the title with _e.g._, `\renewcommand{\bibsection}{}`) to make it look how it is supposed to.
My guess is that it could be fixed in one of the pre/post processing scripts/filters used in Quarto.

**Reply by jjallaire** [reply]
Okay thanks! We'll try to resolve this ASAP.
